paddle/operators/recurrent_op.h

```diff
@@ -167,7 +167,7 @@ class RecurrentGradientAlgorithm {
 
 class RecurrentOp final : public OperatorBase {
 public:
-  void Init() override;
```

Comment:
I see. It seems that we can add `OperatorBase:BeginIteration` in the future if needed.

Why should we move `Init()`'s code into `InferShape` other than into operator's constructor?